### DISCUSSION TITLE
Random Draw thread freeze on "clean" configuration

### DISCUSSION THREAD

**Discussion by navidmafi** [ppy/osu#17098] [Q&A] [answered]
Random freezes in Draw thread, the audio and input are working fine as you can see in the video, i can exit with "Esc"ing back and back and game sound can be heard.

When i was reproducing this, in a rare occasion it freezed the whole system, no Keyboard input. I had to force shutdown.

This video demonstrates the problem i think.

Note : There are some flashing lights occurring at the end. 

https://user-images.githubusercontent.com/46371349/156740769-72575bdb-5ac2-4acb-b515-369cb7ef1767.mp4

Logs:
[DrawFreezeLogs.zip](https://github.com/ppy/osu/files/8184769/DrawFreezeLogs.zip)

Game config: (don't worry i wasn't logged in)
[configs.zip](https://github.com/ppy/osu/files/8184826/configs.zip)

System Information:
![image](https://user-images.githubusercontent.com/46371349/156741848-ee08a3f2-0349-4e4e-a6a7-0a0a72194d4c.png)

Important: i usually put laptop in Eco mode because I don't need Graphics in my workflow.
osu! is running on Intel graphics here/
I don't know if that is contributing to the problem here.

Sorry if i missed something, Ask me and I'll provide it.

**Comment by peppy** [comment]
Sounds like a potential intel driver bug.Does it happen without OBS running?

**Reply by navidmafi** [reply]
Yes it does. OBS was used solely for recording the problem.

**Reply by navidmafi** [reply]
Marking as answer because it seems like the problem does not happen anymore

**Reply by frenzibyte** [reply]
Unmarking answer until a proper solution is provided (coming from https://github.com/ppy/osu/issues/18123).

**Comment by navidmafi** [comment]
Digging through event log i found this. but i don't think i've changed permission settings on any System folder. (assuming that this file really exists)

![image](https://user-images.githubusercontent.com/46371349/156745453-ff359688-e334-4c14-b825-ca083feb259d.png)

**Reply by smoogipoo** [reply]
That one's going to be irrelevant to this issue.

**Comment by smoogipoo** [comment]
Does this always happen? Is it always when entering song select or also when opening e.g. the beatmap listing overlay?

**Reply by navidmafi** [reply]
It does happen nearly always, sometimes even before entering song select on the main menu. 
(For example even scrolling through settings)
It is not specific to opening some menu or page like you said, it's random.
But it mostly happens when entering song select, or entering a map.

**Comment by frenzibyte** [comment]
Upon reading https://github.com/libsdl-org/SDL/issues/5628, @navidmafi can you check what driver version you're currently using, for reference? and whether you've the frame limiter set to `VSync`.

**Reply by navidmafi** [reply]
My frame limiter is set to vsync because 144fps is enough for me considering intel uhd won't give me anything high enough to reach 2xVsync.

Maybe that's why I don't currently have an issue? I believe when I was having issue, my monitor was set to 60Hz and I set my fps cap to 2xVsync

I will try to test with other frame cap settings and refresh rates to see If that's the case as soon as I have the time to do so.

```
GFX Driver Name: Intel® UHD Graphics
Shader Version: 5.1
OpenCL* Version: 3.0
* Microsoft DirectX* *
Runtime Version: 12
Hardware-Supported Version: 12
Vendor ID: 8086
Device ID: 9A68
Device Revision: 01
GFX Driver Version: 27.20.100.9365
Dedicated RAM: 0 GB
Shared RAM: 8 GB
Max Supported Displays: 4
GFX Vulcan: 1.2.170
GFX GOP: 17.0.1061
```

**Reply by frenzibyte** [reply]
Interesting information, @pratham-tharwani can you check your frame limiter settings and try setting it to VSync if you're not already? also, can you fill in your current driver version.

**Reply by pratham-tharwani** [reply]
Driver version is 30.0.100.9805 as reported by Device Manager and Graphics Command Center. Will try setting frame limiter to VSync and report.

**Reply by navidmafi** [reply]
FYI : As soon as I switched from VSync and went though menu and map listings back and forth it freezed with the exact same behavior. 
Will provide more info.

**Comment by navidmafi** [comment, marked as answer]
Thanks to @frenzibyte and Based on my tests, the problem occurred nearly always when not set to VSync (2xVSync - 4x - Unlimited) , no matter what refresh rate was being used.
And Based on playing lazer normally for the past few weeks and having no problem with VSync I can confidently say the problem did not occur using VSync as fps cap.

Some info and thoughts :

- When switching between frame caps, you can reproduce this issue even *before* the main menu loads by setting frame cap with Ctrl+F7 to unlimited. It's almost like the game/draw thread is *guaranteed* to freeze when you set fps cap to Unlimited.

- Based on the previous, the fact that I experienced freeze while switching from main menu to map select, and because blur effect in map select menu is very heavy on GPU, I can see a relationship between how much the GPU is working and how likely it is for the game to freeze.

At the moment, it seems you can have a stable experience by sticking to VSync.

**Reply by peppy** [reply]
Probably the same issue being pointed out in https://github.com/ppy/osu-framework/issues/5164?

**Reply by frenzibyte** [reply]
Along with a workaround request against SDL (https://github.com/libsdl-org/SDL/issues/5628).
